src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -597,6 +653,8 @@ public static function get_merged_data( $origin = 'custom' ) {
 		$result->merge( static::get_user_data() );
 		$result->set_spacing_sizes();
 
+		static::$merged[ $origin ] = $result;
```

Comment:
There are early returns above, so I think this needs to be added to all those clauses as well, otherwise only the `custom` origin will ever have cached values.

Reply:
I actually wonder if there's any value in storing anything other than the completed merged data. I'll raise that question as part of the PR to the Gutenberg repo, but this is good feedback!